packages/mui-base/src/Option/Option.types.ts

```diff
@@ -57,19 +57,11 @@ export interface OptionTypeMap<
 export type OptionProps<
   OptionValue,
   RootComponentType extends React.ElementType = OptionTypeMap<OptionValue>['defaultComponent'],
-> = OverrideProps<OptionTypeMap<OptionValue, {}, RootComponentType>, RootComponentType> & {
-  component?: RootComponentType;
-};
+> = PolymorphicProps<OptionTypeMap<OptionValue, {}, RootComponentType>, RootComponentType>;
 
 export interface OptionType {
```

Comment:
`OptionType` should look similar to `PolymorphicComponent` (but with an additional generic `OptionValue` parameter):

```tsx
export type OptionType = {
  <OptionValue, RootComponentType extends React.ElementType>(
    props: PolymorphicProps<OptionTypeMap<OptionValue>, RootComponentType>,
  ): JSX.Element | null;
  propTypes?: any;
  displayName?: string | undefined;
};
```

Reply:
addressed